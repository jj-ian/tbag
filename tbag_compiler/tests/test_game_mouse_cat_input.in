Wall
Kitchen
Wall
Kitchen
Bedroom