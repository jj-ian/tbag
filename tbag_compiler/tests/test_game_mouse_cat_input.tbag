#import stdlib
#import typeConversionLib

room {}

room Closet { 
	name = "Closet";
}

room Bedroom {
	name = "Bedroom";
}

room Wall {
	name = "Wall";
}

room Kitchen {
	name = "Kitchen";
}

Closet <-> Bedroom;
Closet <-> Wall;
Kitchen <-> Wall;
Kitchen <-> Bedroom;

start { Closet }

npc {
	string roomName;
}

npc Cat {
	roomName = "Bedroom";
}

boolean started = false;

NOT started {
	strPrintLine("You're a mouse.");
	started = true;
}

true {
	printCurrentRoomInfo();
	getInputAdjacentRooms(currentRoom);
	->input
}

currentRoom.name ~~ Cat.roomName {
	print("you got eaten by the cat.\n");
	endgame;
}

func void printCurrentRoomInfo() {
	print("Currently in: ");
	print(currentRoom.name);
	print("\n");
}
