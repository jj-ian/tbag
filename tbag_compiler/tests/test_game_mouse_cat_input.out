You're a mouse.
Currently in: Closet
Choose from one of the following options:
Wall
Bedroom
Input: Wall
Currently in: Wall
Choose from one of the following options:
Kitchen
Closet
Input: Kitchen
Currently in: Kitchen
Choose from one of the following options:
Wall
Bedroom
Input: Wall
Currently in: Wall
Choose from one of the following options:
Kitchen
Closet
Input: Kitchen
Currently in: Kitchen
Choose from one of the following options:
Wall
Bedroom
Input: Bedroom
you got eaten by the cat.